diff --git fbdoom-clean/fbdoom/Makefile fbdoom-workdir/fbdoom/Makefile
index fb32c7b..863822f 100644
--- fbdoom-clean/fbdoom/Makefile
+++ fbdoom-workdir/fbdoom/Makefile
@@ -23,11 +23,7 @@ CC=$(CROSS_COMPILE)gcc  # gcc or g++
 CFLAGS+=-ggdb3 -Os
 LDFLAGS+=-Wl,--gc-sections
 CFLAGS+=-ggdb3 -Wall -DNORMALUNIX -DLINUX -DSNDSERV # -DUSEASM
-LIBS+=-lm -lc
-
-ifneq ($(NOSDL),1)
-	LIBS+= -lSDL
-endif
+LIBS+=
 
 # subdirectory for objects
 OBJDIR=build
@@ -48,8 +44,6 @@ $(OUTPUT):	$(OBJS)
 	@echo [Linking $@]
 	$(VB)$(CC) $(CFLAGS) $(LDFLAGS) $(OBJS) \
 	-o $(OUTPUT) $(LIBS) -Wl,-Map,$(OUTPUT).map
-	@echo [Size]
-	-$(CROSS_COMPILE)size $(OUTPUT)
 
 $(OBJS): | $(OBJDIR)
 
